sdf/compositions/hub.mdx

```diff
@@ -0,0 +1,70 @@
+---
+title: Publishing
+description: Guide to publishing packages to the InfinyOn Hub.
+sidebar_position: 30
+---
+
+# Introduction
+
+The InfinyOn Hub is cloud infrastructure for sharing and discovering SDF package.  Please see [Hub] for more information.
+
+An SDF Package can be published as either a public or private package.  Public packages are available to anyone without restriction.  Private packages are only available to users who have been granted access to the namespace.
+
+By default, all packages are private unless explicitly published as public.
+
+
+
+# Ready for Publishing
+
+SDF Package are immutable.  Once published, the package cannot be changed.  If you need to make changes, you will need to create a new version of the package.  This is why it is important to ensure that the package is tested and ready for publishing.
+
+SDF package follow semver versioning.  The version number is specified in the `meta.version` field of the package file.
+
+Before publishing,
+- Test with `sdf test` to ensure it is working with sample data.
+- Run with test dataflow to ensure it is working in a dataflow.
+
+To test with a dataflow, create a dataflow file and add `dev` section as below: 
+```
+dev:
+  imports:
+    - pkg: example/sentence-pkg@0.1.0
+      path: ./packages/sentence
+```
+
+This will let dataflow use the package from the local directory.
+
+Then run the dataflow with with ephemeral mode: `sdf run --ephemeral` otherwise dataflow will not be able to find the package.
+
+# Publishing
+
+After tested and ready, you can publish the package with `sdf hub publish` command.  Remember, by default, packages are private.  
+
+```bash
+$ sdf hub publish
+```
+
+To list all the packages you can see: `sdf hub list` command.
+
+```bash
+$ sdf hub list
+```
+
+## Publishing as package as public, use `--public` flag
+
+```bash
+$ sdf hub publish --public
+```
+
+
+# Using Hub Package from Dataflow or Package
+
+Please see [Imports] for more information on how to define `imports` in the dataflow or package file that references the package from the Hub.
```

Comment:
how to define -> how to use packages from the Hub in the dataflow or package files.